packages/common/src/utils/stream-utils.ts

```diff
@@ -327,9 +327,27 @@ export class StreamUtils {
         throw new Error(
           `CACAO expired: Commit ${logEntry.cid.toString()} of Stream ${StreamUtils.streamIdFromState(
             state
-          ).toString()} has a CACAO that expired at ${logEntry.expirationTime}. Loading the stream with 'sync: SyncOptions.ALWAYS_SYNC' will restore the stream to a usable state, by discarding the invalid commits (this means losing the data from those invalid writes!)`
+          ).toString()} has a CACAO that expired at ${
+            logEntry.expirationTime
+          }. Loading the stream with 'sync: SyncOptions.ALWAYS_SYNC' will restore the stream to a usable state, by discarding the invalid commits (this means losing the data from those invalid writes!)`
         )
       }
     }
   }
+
+  /**
+   * Given a StreamState returns the timestamp when it was last anchored, or null if the state
+   * hasn't been anchored yet.
+   * @param state
+   */
+  static anchorTimestampFromState(state: StreamState): number | null {
+    for (let i = state.log.length - 1; i >= 0; i--) {
+      const entry = state.log[i]
+      if (entry.timestamp) {
+        return entry.timestamp
+      }
+    }
+
+    return null
+  }
```

Comment:
Any reason we want to return `null` rather than `undefined` here? Either way, this is simpler:
```suggestion
  static anchorTimestampFromState(state: StreamState): number | undefined {
    return state.log.findLast(e => e.timestamp)?.timestamp
  }
```

Reply:
looks like findLast isn't supported in node v16

Reply:
https://app.circleci.com/pipelines/github/ceramicnetwork/js-ceramic/9286/workflows/5b7f5f00-1ea3-4f6a-a470-2c056956b76a/jobs/12266

Reply:
ok, annoying.

Reply:
yeah, support was added in node 18.